src/basic/fd-util.c

```diff
@@ -945,6 +945,13 @@ int path_is_root_at(int dir_fd, const char *path) {
          * kernel is used. In that case, let's assume that we do not have such spurious mount points in an
          * early boot stage, and silently skip the following check. */
 
+        if (check_bind_root &&
+            path_is_root_at_internal(AT_FDCWD, "/", /* check_bind_root = */ false) == 0)
```

Comment:
hmm, this will do a pointless lookup if the original caller invoked `path_is_root_at(AT_FDCWD, "/")` in the first place.

Hence I think we should probably tweak this a bit...

at the top of this function we should maybe add a nice shortcut:

```c
if (check_bind_root && path_is_equal(path, "/"))
        return true;
```

that way for the special case for "/" we'll go from one set of statx calls as before down to zero set of statx calls, and not two!